hail/python/hail/expr/expressions/base_expression.py

```diff
@@ -339,16 +339,36 @@ def describe(self, handler=print):
         handler(s)
 
     def __lt__(self, other):
-        raise NotImplementedError("'<' comparison with expression of type {}".format(str(self._type)))
+        other = to_expr(other)
```

Comment:
Now this stuff is all the same -- can we abstract to:

```python
def _comparison_op(self, op, other):
	other = to_expr(other)
    left, right, success = unify_exprs(self, other)
    if not success:
        raise TypeError(f"Invalid '{op}' comparison, cannot compare expressions "
                        f"of type '{self.dtype}' and '{other.dtype}'")
    return left._bin_op(op, right, hl.tbool)

```

Then we can use this in these new methods, and in in `__eq__` and `__ne__`.


It'll just look like:

```
    def __lt__(self, other):
        return self._comparison_op('!=', other)
```

Reply:
Ah, yeah, totally agree.